File: android-gif-drawable/src/main/java/pl/droidsonroids/gif/GifRenderingExecutor.java
package pl.droidsonroids.gif;

import java.util.concurrent.ScheduledThreadPoolExecutor;

/**
 * Default executor for rendering tasks - {@link java.util.concurrent.ScheduledThreadPoolExecutor}
 * with 1 worker thread and {@link java.util.concurrent.ThreadPoolExecutor.DiscardPolicy}.
 */
final class GifRenderingExecutor extends ScheduledThreadPoolExecutor {

	// Lazy initialization via inner-class holder
	private static final class InstanceHolder {
		private static final GifRenderingExecutor INSTANCE = new GifRenderingExecutor();
	}

	static GifRenderingExecutor getInstance() {
		return InstanceHolder.INSTANCE;
	}

	private GifRenderingExecutor() {
		super(1, new DiscardPolicy());
	}
}

File: android-gif-drawable/src/main/java/pl/droidsonroids/gif/GifRenderingExecutor.kt
package pl.droidsonroids.gif

import java.util.concurrent.ScheduledThreadPoolExecutor

/**
 * Default executor for rendering tasks - [java.util.concurrent.ScheduledThreadPoolExecutor]
 * with 1 worker thread and [java.util.concurrent.ThreadPoolExecutor.DiscardPolicy].
 */
object GifRenderingExecutor : ScheduledThreadPoolExecutor(1, DiscardPolicy())